File: Assets/PolitiRush/Core/Roster.cs
using System.Collections.Generic;
using System.Linq;

namespace PolitiRush.Core
{
    /// <summary>Catalogue des héros et des cibles du niveau par défaut. Miroir de Roster.kt.</summary>
    public static class Roster
    {
        public static readonly List<Hero> Heroes = new List<Hero>
        {
            new Hero("rambeau", "Jean Rambeau", "Bandeau rouge, marcel, jamais de repos.", 6f, 1, 1.2f, Special.Rafale, 14f),
            new Hero("balbo", "Rocco Balbo", "Le boxeur de Philadelphie. Encaisse, puis frappe.", 3f, 3, 1.0f, Special.Uppercut, 12f),
            new Hero("machete", "El Machette", "Il ne texte pas. Il tranche.", 4f, 2, 1.1f, Special.Lames, 15f),
            new Hero("chauve_souris", "Le Chevalier Chauve-Souris", "Milliardaire nocturne, ceinture pleine de gadgets.", 4.5f, 1, 1.3f, Special.Gadget, 16f),
            new Hero("papa_particulier", "Le Papa Très Particulier", "Ex-agent. Il vous trouvera. Il vous tomatera.", 5f, 2, 1.1f, Special.CompetencesParticulieres, 15f),
            new Hero("transporteur", "Le Transporteur Chauve", "Costume impeccable, règles strictes, livraison garantie.", 5f, 1, 1.4f, Special.Livraison, 18f),
        };

        public static Hero Hero(string id) => Heroes.First(h => h.Id == id);

        public static readonly List<PoliticianType> Politicians = new List<PoliticianType>
        {
            PoliticianType.OfTier("prefet", "Le Préfet", Tier.HautFonctionnaire),
            PoliticianType.OfTier("inspecteur_finances", "L'Inspecteur des Finances", Tier.HautFonctionnaire),
            PoliticianType.OfTier("dir_cabinet", "La Directrice de Cabinet", Tier.HautFonctionnaire),
            PoliticianType.OfTier("depute_base", "Le Député de base", Tier.Depute),
            PoliticianType.OfTier("deputee_marche", "La Députée en marche", Tier.Depute),
            PoliticianType.OfTier("depute_insoumis", "Le Député insoumis", Tier.Depute),
            PoliticianType.OfTier("attal", "Gabriel Attal", Tier.Ministre),
            PoliticianType.OfTier("filippetti", "Aurélie Filippetti", Tier.Ministre),
            PoliticianType.OfTier("le_maire", "Bruno Le Maire", Tier.Ministre),
            PoliticianType.OfTier("darmanin", "Gérald Darmanin", Tier.Ministre),
        };

        public static PoliticianType Boss(string id, string name, Tier tier, int hp, int points, int coins) =>
            new PoliticianType(id, name, tier, hp, hp > 100 ? 0.035f : 0.04f, points, coins, true);

        public static readonly List<PoliticianType> Bosses = new List<PoliticianType>
        {
            Boss("macron", "Emmanuel Macron", Tier.ChefEtat, 60, 500, 25),
            Boss("netanyahou", "Benyamin Netanyahou", Tier.ChefEtat, 80, 600, 30),
            Boss("trump", "Donald Trump", Tier.HyperInfluent, 120, 1000, 50),
            Boss("musk", "Elon Musk", Tier.HyperInfluent, 150, 1200, 60),
            Boss("poutine", "Vladimir Poutine", Tier.HyperInfluent, 160, 1300, 60),
            Boss("zuckerberg", "Mark Zuckerberg", Tier.HyperInfluent, 140, 1100, 55),
        };
    }
}

File: core/src/main/kotlin/com/sealkrach/politirush/core/Roster.kt
package com.sealkrach.politirush.core

/**
 * Catalogue des héros et des cibles.
 *
 * Tous les noms sont des pastiches : on reconnaît l'archétype sans utiliser
 * une marque ou une personne réelle. Les inspirations sont documentées dans
 * docs/GAME_DESIGN.md.
 */
object Roster {

    val heroes: List<Hero> = listOf(
        Hero(
            id = "rambeau",
            name = "Jean Rambeau",
            tagline = "Bandeau rouge, mitrailleuse à tomates, jamais de repos.",
            fireRate = 6f,
            damage = 1,
            moveSpeed = 1.2f,
            special = Special.RAFALE,
            specialCooldown = 14f,
        ),
        Hero(
            id = "balbo",
            name = "Rocco Balbo",
            tagline = "Le boxeur de Philadelphie. Encaisse, encaisse, puis frappe.",
            fireRate = 3f,
            damage = 3,
            moveSpeed = 1.0f,
            special = Special.UPPERCUT,
            specialCooldown = 12f,
        ),
        Hero(
            id = "machete",
            name = "El Machette",
            tagline = "Il ne texte pas. Il tranche.",
            fireRate = 4f,
            damage = 2,
            moveSpeed = 1.1f,
            special = Special.LAMES,
            specialCooldown = 15f,
        ),
        Hero(
            id = "chauve_souris",
            name = "Le Chevalier Chauve-Souris",
            tagline = "Milliardaire nocturne, ceinture pleine de gadgets.",
            fireRate = 4.5f,
            damage = 1,
            moveSpeed = 1.3f,
            special = Special.GADGET,
            specialCooldown = 16f,
        ),
        Hero(
            id = "papa_particulier",
            name = "Le Papa Très Particulier",
            tagline = "Ex-agent. Il vous trouvera. Il vous tomatera.",
            fireRate = 5f,
            damage = 2,
            moveSpeed = 1.1f,
            special = Special.COMPETENCES_PARTICULIERES,
            specialCooldown = 15f,
        ),
        Hero(
            id = "transporteur",
            name = "Le Transporteur Chauve",
            tagline = "Costume impeccable, règles strictes, livraison garantie.",
            fireRate = 5f,
            damage = 1,
            moveSpeed = 1.4f,
            special = Special.LIVRAISON,
            specialCooldown = 18f,
        ),
    )

    fun hero(id: String): Hero = heroes.first { it.id == id }

    val politicians: List<PoliticianType> = listOf(
        PoliticianType("promettologue", "Le Promettologue", hp = 2, speed = 0.12f, points = 10, coins = 1),
        PoliticianType("girouette", "La Girouette", hp = 1, speed = 0.20f, points = 15, coins = 1),
        PoliticianType("baron", "Le Baron Local", hp = 4, speed = 0.09f, points = 25, coins = 2),
        PoliticianType("technocrate", "Le Technocrate", hp = 3, speed = 0.11f, points = 20, coins = 2),
        PoliticianType("influenceur", "L'Influenceur Populiste", hp = 2, speed = 0.18f, points = 20, coins = 2),
        PoliticianType("dinosaure", "Le Dinosaure du Sénat", hp = 6, speed = 0.07f, points = 40, coins = 3),
    )

    val bosses: List<PoliticianType> = listOf(
        PoliticianType("candidat_eternel", "Le Candidat Éternel", hp = 60, speed = 0.04f, points = 500, coins = 25, isBoss = true),
        PoliticianType("ministre_reformes", "Le Ministre des Réformes Indispensables", hp = 90, speed = 0.035f, points = 800, coins = 40, isBoss = true),
    )
}
